### PR TITLE
Update to netcoreapp3.0 & bump dependencies

### PR DESCRIPTION
**Pull request by kerams**
WIP

### CODE REVIEW

File: Content/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM microsoft/dotnet:2.2-aspnetcore-runtime-alpine
+FROM microsoft/dotnet:3.0-aspnetcore-runtime-alpine
```
Comment (kerams) [inline Content/Dockerfile]: This image does not seem to exist. Can't pull it from Docker Hub.

File: Content/package.json [hunk 2]
```diff
@@ -1,33 +1,34 @@
...
         "html-webpack-plugin": "^3.2.0",
         "mini-css-extract-plugin": "^0.5.0",
         "open-sans-fonts": "^1.5.1",
-        "react": "^16.9.0",
-        "react-dom": "^16.9.0",
+        "react": "^16.10.2",
+        "react-dom": "^16.10.2",
         "remotedev": "^0.2.9",
-        "sass": "^1.22.10",
+        "resolve-url-loader": "^3.1.0",
```
Comment (kerams) [inline Content/package.json]:
```
Module not found: Error: Can't resolve 'resolve-url-loader' in 'C:\Users\admini\AppData\Local\Temp\2\e714f8fd321748de867231656baf5432'
 @ multi ./src/Client/Client.fsproj ./src/Client/style.scss app[1]
```
without this.

**Comment by kerams** [comment]
```
ERROR in ./.fable/Fable.Elmish.Streams.3.0.3/Program.fs
Module Error (from ./node_modules/fable-loader/index.js):
C:/Users/admini/AppData/Local/Temp/2/e714f8fd321748de867231656baf5432/.fable/Fable.Elmish.Streams.3.0.3/Program.fs(35,31):
(35,39) error FSHARP: The value, constructor, namespace or type 'mbStream' is not defined. (code 39)
```
Not sure what to do with this.

I can't run the tests fully, because they get stuck on
```
Error: listen EADDRINUSE: address already in use 0.0.0.0:8080
    at Server.setupListenHandle [as _listen2] (net.js:1298:14)
    at listenInCluster (net.js:1346:12)
    at doListen (net.js:1485:7)
    at processTicksAndRejections (internal/process/task_queues.js:81:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (net.js:1325:8)
    at processTicksAndRejections (internal/process/task_queues.js:80:21) {
  code: 'EADDRINUSE',
  errno: 'EADDRINUSE',
  syscall: 'listen',
  address: '0.0.0.0',
  port: 8080
}
```
in between individual template permutation tests. Like it does not clean up properly.

File: Content/Dockerfile (hunk 1 above)
Comment (kerams) [reply Content/Dockerfile]: The non-Alpine version is available.

File: Content/src/Server/Server.fsproj [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <Project Sdk="Microsoft.NET.Sdk.Web">
     <PropertyGroup>
         <OutputType>Exe</OutputType>
-        <TargetFramework>netcoreapp2.2</TargetFramework>
+        <TargetFramework>netcoreapp3.0</TargetFramework>
         #if (deploy == "iis" && server != "suave")
```
Comment (kerams) [inline Content/src/Server/Server.fsproj]: This is probably not needed anymore.

**Comment by kerams** [comment]
https://github.com/SAFE-Stack/SAFE-template/pull/318/files#diff-46ee48ad671aebd14ef064487bc1910fL6-L11
might not be needed anymore.

**Comment by kerams** [comment]
At least Giraffe lock files are now lean and mean as expected ;).

File: Content/Dockerfile (hunk 1 above)
Comment (theimowski) [reply Content/Dockerfile]: I think the naming scheme could have changed - looks like there's an alpine image here https://hub.docker.com/_/microsoft-dotnet-core-aspnet/

**Comment by theimowski** [comment]
```
ERROR in ./.fable/Fable.Elmish.Streams.3.0.3/Program.fs
Module Error (from ./node_modules/fable-loader/index.js):
C:/Users/admini/AppData/Local/Temp/2/e714f8fd321748de867231656baf5432/.fable/Fable.Elmish.Streams.3.0.3/Program.fs(35,31):
(35,39) error FSHARP: The value, constructor, namespace or type 'mbStream' is not defined. (code 39)
```

@dbrattli  does it tell you anything?

**Comment by dbrattli** [comment]
@theimowski Yes, let me check ...

File: Content/Dockerfile (hunk 1 above)
Comment (kerams) [reply Content/Dockerfile]: That uses Microsoft's own registry. I'll use that if you're OK with it.

**Comment by kerams** [comment]
Uh, any idea why it still restored 1.2.1 with `FSharp.Control.AsyncRx ~> 1.1.0`? I thought that mean < 1.2

File: Content/src/Client/paket_bridge-fulma-streams.lock [hunk 4]
```diff
@@ -65,42 +65,42 @@ NUGET
...
-      FSharp.Core (>= 4.6.2)
-    Fable.SimpleJson (3.0)
+      FSharp.Core (>= 4.7)
+    Fable.SimpleJson (3.6)
       Fable.Core (>= 3.0)
       Fable.Parsimmon (>= 4.0)
-      FSharp.Core (>= 4.6.2)
-    FSharp.Control.AsyncRx (1.0)
+      FSharp.Core (>= 4.7)
+    FSharp.Control.AsyncRx (1.2.1)
```
Comment (dbrattli) [inline Content/src/Client/paket_bridge-fulma-streams.lock]: @kerams Still pinned in the lock file. You need to `paket update`.

**Comment by kerams** [comment]
Now 1.1 gets restored and compiled by Fable, but still no dice.

**Comment by dbrattli** [comment]
Looks like we need to go back to ~> 1.0

**Comment by kerams** [comment]
Ok, that looks good now. On the other hand, Elmish.Bridge does not like the latest version of Fable.SimpleJson

File: Content/Dockerfile (hunk 1 above)
Comment (theimowski) [reply Content/Dockerfile]: I'm ok with either as long as it's an "official" image

**Comment by theimowski** [comment]
@NHowka @Zaid-Ajaj can you help with that?

**Comment by theimowski** [comment]
@kerams I'd say choose any option as long as it works - let's make this PR pass tests and then make decisions

**Comment by kerams** [comment]
What now?

**Comment by theimowski** [comment]
Seems Travis timeouts - I decreased the number of tests to make it pass.
Let me review the changes and make some local tests now

File: Content/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,5 @@
-FROM microsoft/dotnet:2.2-aspnetcore-runtime-alpine
+FROM mcr.microsoft.com/dotnet/core/aspnet:3.0-alpine
 COPY /deploy /
 WORKDIR /Server
 EXPOSE 8085
-ENTRYPOINT [ "dotnet", "Server.dll" ]
\ No newline at end of file
+ENTRYPOINT [ "./Server" ]
```
Comment (theimowski) [inline Content/Dockerfile]: why this change? I get following error: `standard_init_linux.go:211: exec user process caused "exec format error"`

Comment (kerams) [reply Content/Dockerfile]: .NET Core 3 should produce executables by default, so I thought this would be the way to launch it now.

Comment (kerams) [reply Content/Dockerfile]: Or actually, I get both now. I suppose this line may be reverted then.

**Comment by kerams** [comment]
Elmish.Bridge has been updated, so the Fable.SimpleJson pin can be removed now.

**Comment by Nhowka** [comment]
Sorry, I missed the notification. :sweat_smile:

**Comment by theimowski** [comment]
Ok it looks good to go for me now 👍  let's just wait for CI to pass and I'll merge it

**Comment by theimowski** [comment]
Travis continues to timeout, however I was able to run it [once](https://travis-ci.org/SAFE-Stack/SAFE-template/builds/604941582) successfully 

I'll go and release this and take care of Travis afterwards

Thanks a lot @kerams and @dbrattli for contributing!

**Comment by kerams** [comment]
My pleasure.
